app/ante/vesting.go

```diff
@@ -80,11 +101,35 @@ func (vtd EthVestingTransactionDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx,
 			)
 		}
 
-		// Error if account has locked coins (before surpassing all lockup periods)
-		islocked := clawbackAccount.HasLockedCoins(ctx.BlockTime())
-		if islocked {
-			return ctx, errorsmod.Wrapf(vestingtypes.ErrVestingLockup,
-				"cannot perform Ethereum tx with clawback vesting account, that has locked coins: %s", vested,
+		// Check to make sure that the account does not exceed its spendable balances.
+		// This transaction would fail in processing, so we should prevent it from
+		// moving past the AnteHandler.
+		msgValue := msgEthTx.AsTransaction().Value()
+
+		// Since there can be multiple transactions from different accounts, we track each account's total
+		// spend to compare against its unlocked balances.
+		totalValue, ok := totalValueByAddress[address.String()]
+		if !ok {
+			totalValue = msgValue
+			totalValueByAddress[address.String()] = totalValue
+		} else {
+			totalValue.Add(totalValue, msgValue)
+		}
+
+		// Check that the clawbackAccount has sufficient unlocked tokens to cover all requested spending.
+		ok, lockedBalance := clawbackAccount.LockedCoins(ctx.BlockTime()).Find(evmDenom)
+		if !ok {
+			lockedBalance = sdk.NewCoin(evmDenom, sdk.ZeroInt())
```

Comment:
Q: Why not using a map similarly to the `totalValueByAddress` to avoid repeating the `GetBalance` and `LockedCoins` calls when there're multiple msgs with the same clawbackAcc? or maybe including it on the same map, something like 
```go
map[string]struct {
  totalValue *big.Int, 
  spendableValue *big.Int  
}
```

Reply:
Updated